src/LayerPlan.cpp

```diff
@@ -1698,19 +1703,18 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
 
             gcode.ResetLastEValueAfterWipe(prev_extruder);
 
-            const ExtruderTrain& prev_extruder_train = Application::getInstance().current_slice->scene.extruders[prev_extruder];
-            if (prev_extruder_train.settings.get<bool>("retraction_hop_after_extruder_switch"))
+            const RetractionAndWipeConfig& prev_retraction_config = storage.retraction_wipe_config_per_extruder[prev_extruder];
+            if (prev_retraction_config.retraction_hop_after_extruder_switch)
             {
-                z_hop_height = storage.extruder_switch_retraction_config_per_extruder[prev_extruder].zHop;
-                gcode.switchExtruder(extruder_nr, storage.extruder_switch_retraction_config_per_extruder[prev_extruder], z_hop_height);
+                //TODO: Is it intended that this modifies the Z-Hop down below?
```

Comment:
Yes, see the '// back to normal z hop' comment below, and commit 87dcc26e636aebb74e77388f6f60a2c370b8fe5e

(Hi, finally taking a look at this now. We're getting back on track with picking up PR's after a chaotic year. Sorry for the way too long wait!)